include/deal.II/grid/tria_iterator.h

```diff
@@ -583,21 +585,26 @@ class TriaIterator : public TriaRawIterator<Accessor>
   TriaIterator(const TriaIterator<Accessor> &);
 
   /**
-   * Cross copy constructor from iterators pointing also to non-active
-   * objects.
+   * Conversion constructor from iterators potentially pointing to non-active
+   * objects (i.e., for objects for which we can't tell that the object is
+   * used just by looking at its type).
    *
-   * If the object pointed to is not past-the-end and is not used, the debug
-   * version raises an error!
+   * @pre This constructor will throw an exception if not one of the
+   *   following two conditions are satisfied: (i) the object
+   *   represented by the argument is past-the-end; or (ii) the
+   *   argument represents a used object.
```

Comment:
I had to read `This constructor will throw an exception if not one of the following...` several times before I was convinced I understood what was going on. Could you 'demorganize' this and write a condition starting with `This constructor will throw an exception if...`?

Reply:
> This is basically the text I had originally and that @tjhei did not like and asked me to change.

That is not correct. I complained that you incorrectly used "or" instead of "and". But I agree with David that this is hard to understand. :)

Reply:
How about:
"The argument passed in can be past-the-end or point to a valid object. Otherwise, if it is pointing to an invalid/unused object, an exception will be thrown."